src/reporters/remote_reporter.js

```diff
@@ -57,4 +59,15 @@ export default class RemoteReporter {
         this._sender.close(callback);
         clearInterval(this._intervalHandle);
     }
+
+    setProcess(serviceName: string, tags: Array<Tag>): void {
+        this._process = {
+            'serviceName': serviceName,
+            'tags': ThriftUtils.getThriftTags(tags)
+        };
+
+        if (this._sender) {
```

Comment:
I think reporter's constructor should panic if the sender is null, rather than checking here